### ISSUE TITLE
programatically created BigNumbers are stringified as strings.

### ISSUE THREAD

**Issue by nathanboktae** [sidorares/json-bigint#24]
```
$ node
> var BigNumber = require('bignumber.js')
undefined
> var x = new BigNumber('15283362571593371003')
undefined
> var JSONbig = require('json-bigint')
undefined
> JSONbig.stringify({ x })
'{"x":"15283362571593371003"}'
>
```

It's expected that it should be stringified as `'{"x":15283362571593371003}`

**Comment by sidorares** [comment]
I just tried version from master and behaviour is different: 

```js
var JSONbig = require("./index.js");
var BigNumber = require("bignumber.js");
var x = new BigNumber("15283362571593371003");
console.log(JSONbig.stringify({ x }));
```
outputs  `{"x":15283362571593371003}`
Which version are you using? I'll try to publish current master to npm later today

**Comment by nathanboktae** [comment]
json-bigint@0.2.3
bignumber.js@7.2.1

```
~/tmp
 4:58PM  ᐅ mkdir bignumber-test

~/tmp
 4:58PM  ᐅ cd bignumber-test

~/tmp/bignumber-test
 4:58PM  ᐅ npm install json-bigint bignumber.js
npm WARN saveError ENOENT: no such file or directory, open '/Users/nathanbl/tmp/package.json'
npm WARN enoent ENOENT: no such file or directory, open '/Users/nathanbl/tmp/package.json'
npm WARN tmp No description
npm WARN tmp No repository field.
npm WARN tmp No README data
npm WARN tmp No license field.

+ json-bigint@0.2.3
+ bignumber.js@7.2.1
added 1 package from 1 contributor and updated 2 packages in 1.56s

~/tmp/bignumber-test
 4:58PM  ᐅ node
> var BigNumber = require('bignumber.js')
undefined
> var x = new BigNumber('15283362571593371003')
undefined
> var JSONbig = require('json-bigint')
undefined
> JSONbig.stringify({ x })
'{"x":"15283362571593371003"}'
>
```

There may be a transitive dependency problem if you are using `instanceof` to detect if it's a `BigNumber` or not.

**Comment by sidorares** [comment]
> There may be a transitive dependency problem if you are using instanceof to detect if it's a BigNumber or not.

possibly, in my test BigNumber was the same for lib and test

**Comment by sidorares** [comment]
@nathanboktae no, type check is correct:

```js
    isBigNumber = value != null && (value instanceof BigNumber || BigNumber.isBigNumber(value));

   // which in turn calls
    BigNumber.isBigNumber = function (v) {
      return v instanceof BigNumber || v && v._isBigNumber === true || false;
    };
```

just tried by duplicating `node_modules/bignumber.js` to `node_modules/bignumber2.js`. Even if instanceof chack is falsy `BigNumber.isBigNumber` is correct

**Comment by nathanboktae** [comment]
Yup, that check looks good. But `0.2.3` has this line:

```
            isBigNumber = value != null && (value instanceof BigNumber || value.isBigNumber);;
```

which is not correct. So a new version just needs to be published.

**Comment by sidorares** [comment]
published as v0.3.0
